src/server/pps/server/worker_pool.go

```diff
@@ -177,16 +186,17 @@ func (w *workerPool) discoverWorkers(ctx context.Context) {
 }
 
 func (w *workerPool) addWorker(addr string) error {
-	if cancel, ok := w.workersMap[addr]; ok {
-		cancel()
+	w.workersMapMu.RLock()
```

Comment:
Good point, this wasn't true before when `Cancel` and `Status` were methods on `workerPool` but it is now. I'll remove it.